src/nerve_tools/configure_nerve.py

```diff
@@ -88,23 +79,25 @@ def generate_subconfiguration(service_name, advertise, extra_advertise, port,
 
     # Create a separate service entry for each location that we need to register in.
     for loc, typ in locations_to_register_in:
-        superregions = convert_location_type(loc, typ, 'superregion')
-        for superregion in superregions:
+        zk_locations = convert_location_type(loc, typ, zk_location_type)
+        for zk_location in zk_locations:
             try:
                 zookeeper_topology = get_named_zookeeper_topology(
-                    cluster_type='infrastructure',
-                    cluster_location=superregion
+                    cluster_type=zk_cluster_type,
+                    cluster_location=zk_location,
+                    zk_topology_dir=zk_topology_dir,
                 )
             except:
+                raise
```

Comment:
Whoops! I put this in to figure out a test failure (I had a failure where the output of this function was {} with no helpful guidance). I'll put this back how it was.